### PR TITLE
Remove f16, bf16 and f64 evaluates

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Review by praasz — APPROVED** [review]

File: src/inference/tests/unit/query_model_test.cpp [hunk 1]
```diff
@@ -387,7 +387,7 @@ TEST_F(GetSupportedNodesTest, ShapeOfNonConstantNode) {
     {
         auto param = std::make_shared<ov::op::v0::Parameter>(ov::element::f32, m_shape);
         param->set_friendly_name("input");
-        auto slope_compressed = ov::op::v0::Constant::create(ov::element::f16, ov::Shape{}, {-2.f});
```
Comment (ilya-lavrenov) [inline src/inference/tests/unit/query_model_test.cpp:390]: why do we need to change tests?
I suppose CF should know how to fold f16 constants using convert to f32

File: src/inference/tests/unit/query_model_test.cpp [hunk 2]
```diff
@@ -94,12 +94,12 @@ TEST_F(GetSupportedNodesTest, UnsupportedCompressedConstantCF) {
 
 TEST_F(GetSupportedNodesTest, ConstantSubgraphCF) {
     {
-        auto constant_compressed1 = ov::op::v0::Constant::create(ov::element::f16, m_shape, {1});
+        auto constant_compressed1 = ov::op::v0::Constant::create(ov::element::i16, m_shape, {1});
```
Comment (ilya-lavrenov) [inline src/inference/tests/unit/query_model_test.cpp:97]: should we need as is?

File: src/core/src/op/unsqueeze.cpp [hunk 3]
```diff
@@ -94,10 +94,7 @@ bool evaluate_unsqueeze(const Node* node,
     out->set_shape(out_shape);
 
     using namespace ov::element;
-    return IfTypeOf<i32, i64, u32, u64, f16, f32, f64, bf16>::apply<Evaluate>(element_type,
-                                                                              arg0,
-                                                                              out,
-                                                                              shape_size(out_shape));
+    return IfTypeOf<i32, i64, u32, u64, f32>::apply<Evaluate>(element_type, arg0, out, shape_size(out_shape));
```
Comment (jane-intel) [inline src/core/src/op/unsqueeze.cpp:97]: side note: `Unsqueeze` is a sort of `Reshape`. Maybe it is more meaningful to unify their evaluates. The code doesn't need to be template here.

File: src/core/src/op/eye.cpp [hunk 4]
```diff
@@ -25,8 +25,6 @@ bool evaluate_eye(const ov::HostTensorPtr& out, const int64_t diagonal_index) {
     switch (out->get_element_type()) {
         NGRAPH_TYPE_CASE(evaluate, i8, out, diagonal_index);
         NGRAPH_TYPE_CASE(evaluate, u8, out, diagonal_index);
-        NGRAPH_TYPE_CASE(evaluate, f16, out, diagonal_index);
-        NGRAPH_TYPE_CASE(evaluate, bf16, out, diagonal_index);
         NGRAPH_TYPE_CASE(evaluate, i32, out, diagonal_index);
         NGRAPH_TYPE_CASE(evaluate, f32, out, diagonal_index);
         NGRAPH_TYPE_CASE(evaluate, f64, out, diagonal_index);
```
Comment (jane-intel) [inline src/core/src/op/eye.cpp:30]:
```suggestion
```

**Review by jane-intel — APPROVED** [review]
My prediction -- no impact from f64 removal on ShapeOf sub-graphs evaluation.

**Pull request closed without merging**